supabase/functions/brain_v1_1/_shared/supabaseClient.ts

```diff
@@ -0,0 +1,12 @@
+import { createClient } from 'https://esm.sh/@supabase/supabase-js@2.20.0'
+
+let supabase_url = Deno.env.get('Z_SUPABASE_URL') ?? ''
+let supabase_anon_key = Deno.env.get('Z_SUPABASE_ANON_KEY') ?? ''
+
+const checkDeployed = Deno.env.get('DENO_DEPLOYMENT_ID') ?? ''
+if (checkDeployed.startsWith('tpqbderafyftvmrhrdht')) {
+  supabase_url = Deno.env.get('SUPABASE_URL') ?? supabase_url
```

Comment:
What's the difference between ``Z_SUPABASE_URL`` and ``SUPABASE_URL`` env vars? Same question for the anon key

Reply:
good question - supabase local dev studio takes care of giving you `SUPABASE_URL` but not the correct one if you're doing local work, same with the anon key. however, it is correct in production for a deployed edge function. 

to keep things consistent regardless of production environment i use Z_. open to renaming to `PERSISTED_SUPABASE_URL` and `PERSISTED_SUPABASE_ANON_KEY` or something better

Reply:
What do you mean by "supabase local dev studio takes care of giving you SUPABASE_URL but not the correct one if you're doing local work"? I mean, you can specify the ``.env`` file, right?

Reply:
when running supabase locally via CLI and docker instances, it will take care of setting some supabase environment variables like SUPABASE_URL (to http://supabase_kong_dubdubs:8000), SUPABASE_ANON_KEY, etc... (run `supabase status` in terminal to see these). im not interested in setting up supabase studio and all that, i just want to read/write to our deployed DB.

i dont want to neglect that env vars supabase sets for us automatically, so if the edge function is being executed in prod, let's use it! otherwise default to permanent environment variable. would it be simpler to just use `Z_SUPABASE` for both all the time and remove the extra LOC, yes. but i wanted to be safe and keep the prod supabase client **always** correct by referencing the supabase generated env vars.